src/lib/utils/meta_tables/meta_system_utilization_table.cpp

```diff
@@ -0,0 +1,353 @@
+                                               {"load_average_15_min", DataType::Float, false},
+                                               {"system_ram_total", DataType::Long, false},
+                                               {"system_ram_free", DataType::Long, false},
+                                               {"process_virtual_memory", DataType::Long, false},
+                                               {"process_physical_memory", DataType::Long, false}}) {}
+
+const std::string& MetaSystemUtilizationTable::name() const {
+  static const auto name = std::string{"system_utilization"};
+  return name;
+}
+
+void MetaSystemUtilizationTable::init() {
+  _get_system_cpu_usage();
+  _get_process_cpu_usage();
+}
+
+std::shared_ptr<Table> MetaSystemUtilizationTable::_on_generate() {
+  auto output_table = std::make_shared<Table>(_column_definitions, TableType::Data, std::nullopt, UseMvcc::Yes);
+
+  const auto system_cpu_usage = _get_system_cpu_usage();
+  const auto process_cpu_usage = _get_process_cpu_usage();
+  const auto load_avg = _get_load_avg();
+  const auto system_memory_usage = _get_system_memory_usage();
+  const auto process_memory_usage = _get_process_memory_usage();
+
+  output_table->append({system_cpu_usage, process_cpu_usage, load_avg.load_1_min, load_avg.load_5_min,
+                        load_avg.load_15_min, system_memory_usage.total_ram, system_memory_usage.free_ram,
+                        process_memory_usage.virtual_memory, process_memory_usage.physical_memory});
+
+  return output_table;
+}
+
+MetaSystemUtilizationTable::LoadAvg MetaSystemUtilizationTable::_get_load_avg() {
+#if defined __linux__
+
+  std::ifstream load_avg_file;
+  load_avg_file.open("/proc/loadavg", std::ifstream::in);
+
+  std::string load_avg_value;
+  std::vector<float> load_avg_values;
+  for (int value_index = 0; value_index < 3; ++value_index) {
+    std::getline(load_avg_file, load_avg_value, ' ');
+    load_avg_values.push_back(std::stof(load_avg_value));
+  }
+  load_avg_file.close();
+
+  return {load_avg_values[0], load_avg_values[1], load_avg_values[2]};
+
+#elif defined __APPLE__
+
+  loadavg load_avg;
+  size_t size = sizeof(load_avg);
+  if (sysctlbyname("vm.loadavg", &load_avg, &size, nullptr, 0) != 0) {
+    Fail("Unable to call sysctl vm.loadavg");
+  }
+
+  return {static_cast<float>(load_avg.ldavg[0]) / static_cast<float>(load_avg.fscale),
+          static_cast<float>(load_avg.ldavg[1]) / static_cast<float>(load_avg.fscale),
+          static_cast<float>(load_avg.ldavg[2]) / static_cast<float>(load_avg.fscale)};
+
+#endif
+
+  Fail("Method not implemented for this platform");
+}
+
+int MetaSystemUtilizationTable::_get_cpu_count() {
+#if defined __linux__
+
+  std::ifstream cpu_info_file;
+  cpu_info_file.open("/proc/cpuinfo", std::ifstream::in);
+
+  uint32_t processors = 0;
+  for (std::string cpu_info_line; std::getline(cpu_info_file, cpu_info_line);) {

```

Comment:
I'm not a fan of for loops that do not follow the for loop pattern (loop variable etc). This looks much more like a while loop to me.